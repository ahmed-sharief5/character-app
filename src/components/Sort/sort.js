import React from "react";
import PropTypes from 'prop-types';

const Sort = ({ onClick, byWhich, sortBy }) => {
    return (
        <button type="button" onClick={onClick} className="btn btn-info">{`Sort ${byWhich} by ${sortBy}`}</button>
    );
}

Sort.propTypes = {
    onClick: PropTypes.func,
    byWhich: PropTypes.string,
    sortBy: PropTypes.string
}

export default Sort;