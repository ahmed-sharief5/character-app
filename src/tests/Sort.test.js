import React from "react";
import Sort from "../components/Sort";
import renderer from "react-test-renderer";

it("renders sort component correctly", () => {
    const tree = renderer.create(<Sort/>).toJSON();
    expect(tree).toMatchSnapshot();
});
