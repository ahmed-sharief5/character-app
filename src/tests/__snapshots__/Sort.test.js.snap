// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders sort component correctly 1`] = `
<button
  className="btn btn-info"
  type="button"
>
  Sort undefined by undefined
</button>
`;
